xocto/ranges.py

```diff
@@ -115,6 +115,8 @@ class Range(Generic[T]):
         None
         >>> Range(0, 2) | Range(2, 4, boundaries="(]")
         None
+        >>> Range(0, 2, boundaries="[]") | Range(3, 4)
+        None
```

Comment:
I'd say this example needs a justification. What is the learning here about using `Range` with integers?

Reply:
I agree but I have no idea how to word it.

Reply:
Put it at the top, something like

"Note that using `Range` with `date` objects does not convert them to `datetimes`. Thus the end of one date does not overlap with the beginning of the next date like it might in an end-inclusive `datetime` comparison. To achieve this, convert your dates to `datetimes` before creating the range.

Reply:
I'm not sure that's a full description of the issue.

I think it's more something like - the documentation on `union` says that adjacent ranges will be merged, but that's misleading.

So I think the right wording is something to do with a the meaning of _adjacent_ - what it really means is that the ranges have to either start and end on exactly the same value, or overlap a bit. They can't just inclusively touch values which are next to each other.

For example if we had one range which ended (inclusive) 1 millisecond* before midnight, and another which started (inclusive) on exactly midnight, they wouldn't be considered adjacent. They can only be considered adjacent if they're exactly the same value.

*Nanosecond? Whatever the smallest increment is.

Reply:
@cogat I've updated with a 2nd attempt.